src/elevator/ElevatorSubSystem.java

```diff
@@ -90,24 +95,12 @@ public void handleMessage(Message m) {
 	}
 
 	private void sendTravelMessage() {
-		FloorTravelMessage m = new FloorTravelMessage();
-		m.setElevator(elevator.getId());
-		m.setStartingFloor(elevator.getFloor());
-
-		int diff = 1;
 		if (elevator.getState() == Elevator.State.MOVING_UP) {
-			m.setDirection(Direction.UP);
+			arrivalSensor = new Thread(new ArrivalSensor(elevator.getId(), elevator.getFloor(), elevator.getFloor()+1));
```

Comment:
Can you explain why this is creating a new arrivalSensor instead of creating a new message and sending it to the existing sensor?

Reply:
The arrivalSensor thread terminates once it sends the arrival message; its not like server. I'm open to changing this though.